server/mergin/sync/workspace.py

```diff
@@ -281,3 +281,36 @@ def update_project_members(
     def access_requests_query():
         """Project access base query"""
         return AccessRequest.query.join(Project)
+
+    def project_access(self, project: Project) -> List[MemberAccess]:
+        """
+        Project access users overview
+        """
+        ws = self.factory_method()
+        if (
+            Configuration.GLOBAL_ADMIN
+            or Configuration.GLOBAL_WRITE
+            or Configuration.GLOBAL_READ
+        ):
+            members = User.query.filter(User.active.is_(True)).all()
+        else:
+            member_ids = set(
+                project.access.readers + project.access.writers + project.access.owners
+            )
+            members = User.query.filter(User.active, User.id.in_(member_ids)).all()
+        result = []
+        for member in members:
+            member_access = MemberAccess(
+                id=member.id,
+                username=member.username,
+                role=ws.get_user_role(member),
+                name=member.profile.name(),
+                email=member.email,
+            ).to_dict()
+            # add correct project permission by comparing ws role and project access permission
+            user = User.query.get(member.id)
```

Comment:
in case of global members this would cause lot of calls to db

I would probably split lists to global_members and project_members and do some magic there:
- if global_members then filter from them project_members and use it for role override
- else filter only project_members from db and use its role directly

Reply:
maybe this query is not needed (I just wanted to be safe) 
Tests are happy when I pass `member` object to `get_user_project_role()`